lib/libconfig.c

```diff
@@ -839,6 +847,7 @@ static config_setting_t *config_setting_create(config_setting_t *parent,
   setting->config = parent->config;
   setting->hook = NULL;
   setting->line = 0;
+  setting->comment = (comment == NULL) ? NULL : strdup(comment);
```

Comment:
__config_setting_destroy() needs to be updated to free this string, otherwise we have a memory leak